src/db.rs

```diff
@@ -1719,10 +1719,11 @@ mod tests {
         let fp_registry = Arc::new(FailPointRegistry::new());
         let object_store: Arc<dyn ObjectStore> = Arc::new(InMemory::new());
         let path = Path::from("/tmp/test_kv_store");
-        let kv_store = Db::open_with_opts(
+        let kv_store = Db::open_with_fp_registry(
```

Comment:
oof. we should add the inverse to the test case - e.g. doing a read with level `Committed` and asserting it returns `None`, which would have failed before this test.

Reply:
I think that's tested in the following test (`test_should_read_only_committed_data`). It checks both read committed and uncommitted writes.

Reply:
right, but here it would be done as a safeguard to ensure we're testing what we think we're testing, which we weren't previously because I forgot to pass in the registry.

Reply:
Ah, gotcha. Updated the PR to add a paranoia assertion. I also validated that the code sometimes fails the old way if I enable multiple threads.